bootstrap/shell/v002__update_defaults.sh

```diff
@@ -0,0 +1,263 @@
+  run_cmd $cmd
+}
+
+function put_topology_component_bundle {
+  uri=$1
+  data=$2
+  subType=$3
+  out=$(curl -s -X GET -H "Content-Type: application/json" -H "Cache-Control: no-cache" "${CATALOG_ROOT_URL}$uri?subType=${subType}&streamingEngine=STORM")
+  bundleId=$(getId $out)
+  cmd="curl -i --negotiate -u:anyUser  -b /tmp/cookiejar.txt -c /tmp/cookiejar.txt -sS -X PUT -i -F topologyComponentBundle=@$data ${CATALOG_ROOT_URL}$uri/$bundleId"
+  echo "PUT $data"
+  run_cmd $cmd
+}
+
+function put_service_bundle {
+  uri=$1
+  data=$2
+  cmd="curl -i --negotiate -u:anyUser  -b /tmp/cookiejar.txt -c /tmp/cookiejar.txt -sS -X PUT ${CATALOG_ROOT_URL}$uri --data @$data -H 'Content-Type: application/json'"
+  echo "PUT $data"
+  run_cmd $cmd
+
+}
+
+#Below command to update storm version will be called by RE script. Need to remove later. Adding now for convenience
+update_storm_version_command="$bootstrap_dir/update-storm-version.sh 1.1.0.3.0.0.0-453"
+run_cmd $update_storm_version_command
+
+#---------------------------------------------
+# Get catalogRootUrl from configuration file
+#---------------------------------------------
+
+CONF_READER_MAIN_CLASS=com.hortonworks.registries.storage.tool.sql.PropertiesReader
+
+for file in "${bootstrap_dir}"/lib/*.jar;
+do
+    CLASSPATH="$CLASSPATH":"$file"
+done
+
+CATALOG_ROOT_URL_PROPERTY_KEY=catalogRootUrl
+component_dir=${bootstrap_dir}/components
+service_dir=${bootstrap_dir}/services
+user_role_dir=${bootstrap_dir}/users_roles
+
+echo "Configuration file: ${CONFIG_FILE_PATH}"
+
+CATALOG_ROOT_URL=`exec ${JAVA} -cp ${CLASSPATH} ${CONF_READER_MAIN_CLASS} ${CONFIG_FILE_PATH} ${CATALOG_ROOT_URL_PROPERTY_KEY}`
+
+# if it doesn't exit with code 0, just give up
+if [ $? -ne 0 ]; then
+  exit 1
+fi
+
+echo "Catalog Root URL: ${CATALOG_ROOT_URL}"
+echo "Component bundle Root dir: ${component_dir}"
+echo "Service bundle Root dir: ${service_dir}"
+echo "User/Role bundle Root dir: ${user_role_dir}"
+
+function update_bundles {
+    # === Source ===
+    # === Processor ===
+    add_topology_component_bundle /streams/componentbundles/PROCESSOR ${component_dir}/processors/v001__rtjoin-bolt-topology-component.json
+    # === Sink ===
+    put_topology_component_bundle /streams/componentbundles/SINK ${component_dir}/sinks/v002__hdfs-sink-topology-component.json HDFS
+    put_topology_component_bundle /streams/componentbundles/SINK ${component_dir}/sinks/v002__jdbc-sink-topology-component.json JDBC
+    put_topology_component_bundle /streams/componentbundles/SINK ${component_dir}/sinks/v002__hive-sink-topology-component.json HIVE
+    # === Topology ===
+    # === Service Bundle ===
+    put_service_bundle /servicebundles/KAFKA ${service_dir}/v002__kafka-bundle.json
+    put_service_bundle /servicebundles/STORM ${service_dir}/v002__storm-bundle.json
+    put_service_bundle /servicebundles/ZOOKEEPER ${service_dir}/v002__zookeeper-bundle.json
+    post /servicebundles ${service_dir}/v001__druid-bundle.json
```

Comment:
should this be in the v001__ingest_defaults.sh?

Reply:
@raju-saravanan +1, looks great.

Can you raise an issue/JIRA to track the changes for Ambari upgrades as well?